morpholopy/galaxy_plots.py

```diff
@@ -65,12 +66,54 @@
 bingrid = 0.8 * unyt.kpc
 npix_coarse = int(2.0 * r_img_kpc / bingrid)
 
-cmap = plt.get_cmap("Greens")
+cmap = pl.get_cmap("Greens")
 usergreen = cmap(0.7)
-cmap = plt.get_cmap("Blues")
+cmap = pl.get_cmap("Blues")
 userblue = cmap(0.7)
 
 
+def exponential(x, Sigma0, H, offset):
+    return Sigma0 * np.exp(-np.abs(x + offset) / H)
+
+def calculate_scaleheight_pointmasses(zcoords, weights, zrange, resolution):
+
+    S_1D, bin_edges = np.histogram(zcoords, bins = resolution, range = (-zrange, zrange),
+                        weights = weights, density = False)
+
+    z_1D = (bin_edges[1:] + bin_edges[:-1])/2.
+    p0 = (hist.max(), 1., 0.)
+
+    try:
+        popt, pcov = curve_fit(
+                exponential, z_1D[np.isfinite(S_1D)], S_1D[np.isfinite(S_1D)]
+        )
+    except:
+        return np.nan
+
+    return popt[1]
+        
+
+def calculate_scaleheight_fit(mass_map, r_img_kpc, r_abs_max_kpc):
+    xx = np.linspace(
+        -r_img_kpc.value, r_img_kpc.value, len(mass_map[:, 0]), endpoint=True
+    )
+    z = (np.tile(xx, (len(xx), 1))).T
+    z_1D = np.ravel(z[:, (np.abs(xx) < r_abs_max_kpc)])
+    S_1D = np.ravel(mass_map[:, (np.abs(xx) < r_abs_max_kpc)])
+
+    p0 = (mass_map.max(), 1., 0.)
+
+    try:
+        popt, pcov = curve_fit(
+                exponential, z_1D[np.isfinite(S_1D)], S_1D[np.isfinite(S_1D)],
+                bounds = ( (0., 1.e-5, -5.), (np.inf, 100., +5.) )
+        )
+    except:
+        return np.nan
```

Comment:
Same as above.

Reply:
As above. I don't have a strong preference as long as we only plot values where we did get a reasonable fit.